### DISCUSSION TITLE
Now I am having error in backend.yml

### DISCUSSION THREAD

**Discussion by Priya-Devops2023** [learndevopsonline/learndevopsonline#1852] [Q&A]
**TASK [backend : Configure Backend Service] *************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'env' is undefined. 'env' is undefined. 'env' is undefined. 'env' is undefined
fatal: [backend-dev.pdp2023.online]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'env' is undefined. 'env' is undefined. 'env' is undefined. 'env' is undefined"}**

PLAY RECAP *********************************************************************
backend-dev.pdp2023.online : ok=4    changed=2    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

23.23.11.136 | 172.31.46.173 | t3.small | https://github.com/Priya-Devops2023/expense-ansible.git

There is no env variable declared in the code.

Kindly let me know whats the issue.
Thanks in advance,
Priya

**Comment by Priya-Devops2023** [comment]
This is the code in backend.service

[Unit]
Description = Backend Service

[Service]
User=expense
Environment=DB_HOST="mysql-dev.pdp2023.online"
ExecStart=/bin/node /app/index.js
SyslogIdentifier=backend

[Install]
WantedBy=multi-user.target

**Comment by chaithanya1812** [comment]
![image](https://github.com/learndevopsonline/learndevopsonline/assets/111736742/dc51d115-d846-4a64-883f-b20a4fdff644)

## Next while posting an error message display in screenshots like this ~~

![image](https://github.com/learndevopsonline/learndevopsonline/assets/111736742/5dd6abcb-d25f-4f12-8a55-d39fcc4b169e)

**Comment by Priya-Devops2023** [comment]
Hi Chaithanya,
   sure .
     still the error presists

![image](https://github.com/learndevopsonline/learndevopsonline/assets/154531102/4d10810b-c304-43c8-ac2d-9f96beabe9c1)

![image](https://github.com/learndevopsonline/learndevopsonline/assets/154531102/ecd3c025-0efa-428f-ad15-5e6dea6eafad)


Thanks in advance
-Priya

**Comment by chaithanya1812** [comment]
# Issue 1
![image](https://github.com/learndevopsonline/learndevopsonline/assets/111736742/d4474e0f-565c-4117-b820-0c7d345402d1)

# Issue 2

you are using variable called  < env > define it..

**Reply by Priya-Devops2023** [reply]
Hi Chaitanya

   Its working now.Thanks for the help.
    In code i changed "template" to "copy" command it worked and changed the directory also from templates to files.

Thanks 
Priya.
